sycl/doc/extensions/experimental/sycl_ext_oneapi_device_global.asciidoc

```diff
@@ -324,6 +326,21 @@ The object of type `T` is initialized from the `args` parameter pack using list
 
 `T` must be trivially destructible.
 
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+constexpr device_global(const device_global &other);
```

Comment:
Am I right in thinking that this copy constructor will only work if both device globals have the same template arguments (i.e., the same types and property lists)?

If so, that could potentially be surprising:

```c++
// works
int init = 3;
float copy = init;

// breaks?
oneapiext::device_global<const int> DGInit{3};
oneapiext::device_global<const float> DGCopy{DGInit};
```

With properties, there could also be cases where the variables have the same type but different properties (for optimization purposes).  This feels like a can of worms, because I think we'd have to define where this copy constructor executes in order to make sense of the `host_access` properties....

I'm sorry for making this more complicated, but I'm asking this because I'm not clear on the use-case for this.  If users are asking for device globals to behave more like regular globals, I think we might at least need to address the type conversion part.

Reply:
I tried it a bit, but the problem becomes that it then prefers the consteval variadic ctor over templated copy ctors. I can give it another shot though.

Reply:
I have made some changes so that differing template arguments are now allowed as well.

> This feels like a can of worms, because I think we'd have to define where this copy constructor executes in order to make sense of the host_access properties....

I don't think there's a big problem there. For `host_access` we already cheat a little when `device_image_scope` is _not_ there, but it's an implementation detail so I don't think the user should have to worry.

Reply:
Thanks, the new overload looks good.  Just one question about the description of the constraints.